lib/Runtime/PlatformAgnostic/Platform/Linux/UnicodeText.ICU.cpp

```diff
@@ -101,13 +113,172 @@ namespace PlatformAgnostic
+            }
+
+            if (charType == U_COMBINING_SPACING_MARK)
+            {
+                return UnicodeGeneralCategoryClass::CategoryClassSpacingCombiningMark;
+            }
+
+            if (charType == U_NON_SPACING_MARK)
+            {
+                return UnicodeGeneralCategoryClass::CategoryClassNonSpacingMark;
+            }
+
+            if (charType == U_CONNECTOR_PUNCTUATION)
+            {
+                return UnicodeGeneralCategoryClass::CategoryClassConnectorPunctuation;
+            }
+            
+            return UnicodeGeneralCategoryClass::CategoryClassOther;
+        }
+
+        // We actually don't care about the legacy behavior on Linux since no one
+        // has a dependency on it. So this actually is different between
+        // Windows and Linux
+        CharacterClassificationType GetLegacyCharacterClassificationType(char16 character)
+        {
+            auto charTypeMask = U_GET_GC_MASK(character);
+
+            if ((charTypeMask & U_GC_L_MASK) != 0)
+            {
+                return CharacterClassificationType::Letter;
+            }
+            
+            if ((charTypeMask & (U_GC_ND_MASK | U_GC_P_MASK)) != 0)
+            {
+                return CharacterClassificationType::DigitOrPunct;
+            }
+
+            // As per http://archives.miloush.net/michkap/archive/2007/06/11/3230072.html
+            //  * C1_SPACE corresponds to the Unicode Zs category.
+            //  * C1_BLANK corresponds to a hardcoded list thats ill-defined.
+            // We'll skip that compatibility here and just check for Zs.
+            if ((charTypeMask & U_GC_ZS_MASK) != 0)
+            {
+                return CharacterClassificationType::Whitespace;
+            }
+            
+            return CharacterClassificationType::Invalid;
+        }
+            
+        uint32 ChangeStringLinguisticCase(CaseFlags caseFlags, const char16* sourceString, uint32 sourceLength, char16* destString, uint32 destLength, ApiError* pErrorOut)
+        {
+            int32_t resultStringLength = 0;
+            UErrorCode errorCode = U_ZERO_ERROR;
+
+            static_assert(sizeof(UChar) == sizeof(char16), "Unexpected char type from ICU, function might have to be updated");
+            if (caseFlags == CaseFlagsUpper)
+            {
+                resultStringLength = u_strToUpper((UChar*) destString, destLength,
+                    (UChar*) sourceString, sourceLength, NULL, &errorCode);
+            }
+            else if (caseFlags == CaseFlagsLower)
+            {
+                resultStringLength = u_strToLower((UChar*) destString, destLength,
+                    (UChar*) sourceString, sourceLength, NULL, &errorCode);
+            }
+            else
+            {
+                Assert(false);
+            }
+
+            if (U_FAILURE(errorCode))
+            {
+                *pErrorOut = TranslateUErrorCode(errorCode);
+                return -1;
```

Comment:
Done- I had originally defined ChangeStringLinguisticCase on Windows as uint- I don't remember why, even the Windows variant works fine with the int32 return type, so changed it.